tools/sim/bridge/metadrive.py

```diff
@@ -0,0 +1,130 @@
+import warnings
+import numpy as np
+import math
+
+from tools.sim.bridge.common import World, SimulatorBridge, Camerad, STEER_RATIO, W, H
+from tools.sim.bridge.utils import timer
+
+class MetaDriveWorld(World):
+  def __init__(self, env, ticks_per_frame: float):
+    self.env = env
+    self.speed = 0.0
+    self.yuv = None
+    self.camerad = Camerad()
+    self.ticks_per_frame = ticks_per_frame
+
+  def apply_controls(self, steer_sim, throttle_out, brake_out, rk):
+    vc = [0.0, 0.0]
+    vc[0] = steer_sim * -1
+    if throttle_out:
+      vc[1] = throttle_out * 10
+    else:
+      vc[1] = -brake_out
+    if rk.frame % self.ticks_per_frame == 0:
+      o, _, d, _ = timer("Step", lambda: self.env.step(vc))
+      if d:
+        print("!!!Episode terminated due to violation of safety!!!")
+        self.env.reset()
+        self.env.step([0.0, 0.0])
+        for _ in range(300):
+          rk.keep_time()
+      
+      self.speed = o["state"][3] * math.sqrt(self.ticks_per_frame) * 10 # empirically derived
+
+      # if (rk.frame % (self.ticks_per_frame * 2)) == 0:
+      img = timer("pixel array", lambda: self.env.vehicle.image_sensors["rgb_wide"].get_pixels_array(self.env.vehicle, False))
+      self.yuv = timer("yuv kernel", lambda: self.camerad.img_to_yuv(img))
+      timer("send yuv", lambda: self.camerad.cam_send_yuv_wide_road(self.yuv))
+      # timer("send yuv", lambda: self.camerad.cam_send_yuv_wide_road(self.yuv))
+
+  def get_velocity(self):
+    return None
+
+  def get_speed(self) -> float:
+    return self.speed
+
+  def get_steer_correction(self) -> float:
+    max_steer_angle = 75 / self.ticks_per_frame
+    return max_steer_angle * STEER_RATIO * -1
+  
+  def tick(self):
+    pass
+
+
+class MetaDriveBridge(SimulatorBridge):
+  TICKS_PER_FRAME = 10
+
+  def __init__(self, args):
+    if args.dual_camera:
+      warnings.warn("Dual camera not supported in MetaDrive simulator for performance reasons")
```

Comment:
better to exit in this case

Reply:
We could probably support this for dGPUs. I think we should multiplex on whether we detect CUDA or ROCm. (don't think we'll care about Xe for now...)

Reply:
(well, we can just piggyback on `--high_quality` for now I suppose. But auto multiplexing the default would be nice.)

Reply:
Will multiplex throwing an error based on `high_spec`. And anw, `high_spec` is already used to multiplex the dual camera for CARLA.

I suppose `high_spec` may have a different meaning depending on whether one chooses the metadrive or carla simulator.

I'm not sure we want simulator-specific config for the time being, don't want running it to be too complicated.

Reply:
Made an error, we will implement it once we get this bare-bones version committed.